csrc/device_lower/pass/index.cpp

```diff
@@ -1479,8 +1520,21 @@ void IndexLowering::handle(const PadOp* pad) {
 
   const auto pad_val = pad->value();
 
+  std::unordered_map<IterDomain*, Val*> override_index;
+  for (auto padded_axis : pad->getPaddedAxes()) {
+    auto padded_id = producer_doms.at(padded_axis);
+    if (padded_id->isBroadcast()) {
+      // When we pad a Broadcast IterDomain, we should not treat it as a
+      // Broadcast as we normally would. Instead, we will treat it as a regular
+      // Iteration domain with extent 1.
+      auto ind =
```

Comment:
Instead of copy-pasting a function, does it make sense to make a clone of the tensor domain, and use 
https://github.com/NVIDIA/Fuser/blob/fb0e25e50f427305972de9d96dda21fa3606c5d7/csrc/device_lower/utils.h#L47
to temporarily modify the tensor domain?

Reply:
Oh I hadn't even considered doing that. Let me try it out!

Reply:
This method is slightly complicated since we need to create a replacement `IterDomain` for each broadcast `IterDomain` in the producer's rfactor domain that is being padded, and that replacement must be derived from the root domain or else we get an error when trying to create a replacement `TensorDomain`. That means we then need to actually replace all the siblings of each padded broadcast ID, and so on.